File: frontend/lib/models/budget_proposal.dart
class BudgetProposal {
  final int id;
  final int projectId;
  final String companyName;
  final double amount;
  final int votes;
  final int totalCitizens;

  BudgetProposal({
    required this.id, required this.projectId, required this.companyName,
    required this.amount, required this.votes, required this.totalCitizens
  });

  factory BudgetProposal.fromJson(Map<String, dynamic> json) {
    return BudgetProposal(
      id: json['id'],
      projectId: json['projectId'],
      companyName: json['companyName'] ?? 'Empresa',
      amount: json['amount'] ?? 0.0,
      votes: json['votes'] ?? 0,
      totalCitizens: json['totalCitizensInLocality'] ?? 100,
    );
  }
  
  double get percentage => totalCitizens == 0 ? 0 : votes / totalCitizens;
  bool get isWinner => percentage >= 0.8;
}

File: backend/src/main/java/com/civicworks/model/mysql/BudgetProposal.java
package com.civicworks.model.mysql;
import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;

@Entity
@Data
@Table(name = "budget_proposals")
public class BudgetProposal {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private Long projectId;
    private Long companyId;
    private String companyName;
    
    @Column(precision = 19, scale = 4)
    private BigDecimal amount;
    
    private String description;
    
    private int votes;
    
    @Transient
    private long totalCitizensInLocality; 
}
